pkg/services/stateroot/service.go

```diff
@@ -26,7 +26,7 @@ type (
 		GetConfig() config.Blockchain
 		GetDesignatedByRole(role noderoles.Role) (keys.PublicKeys, uint32, error)
 		HeaderHeight() uint32
-		SubscribeForBlocks(ch chan *block.Block)
+		SubscribeForBlocks(ch chan<- *block.Block)
```

Comment:
We have `SubscribeForNotaryRequests`, the same should be done with this API in a separate commit. And the same thing should be done for mempool subscriptions (`SubscribeForTransactions`).

Reply:
Isn't  this func (s *Server) SubscribeForNotaryRequests(ch chan<- mempoolevent.Event) already send-only?

Reply:
Turns out it is, so nothing should be done wrt this.